### PR TITLE
Add buttons to load defaults

### PR DESCRIPTION
**Pull request by TreTuna**
As a solution to https://github.com/keyboardio/Chrysalis/issues/362 this adds buttons on the import/export component that allow easy loading of the default layouts.

add-default-buttons

Signed-off-by: Tré Ammatuna <tre@tretuna.com>

### CODE REVIEW

File: src/renderer/screens/Editor/ImportExportDialog.js [hunk 1]
```diff
@@ -81,16 +105,30 @@ export const ImportExportDialog = withSnackbar(props => {
...
             alignItems: "center"
           }}
         >
-          <CopyToClipboard text={data} onCopy={onCopySuccess}>
-            <Button color="primary">{i18n.editor.copyToClipboard}</Button>
-          </CopyToClipboard>
-          <Button color="primary" onClick={pasteFromClipboard}>
-            {i18n.editor.pasteFromClipboard}
-          </Button>
+          <div style={{ display: "flex" }}>
```
Comment (algernon) [inline src/renderer/screens/Editor/ImportExportDialog.js]: Since the defaults are only available for the Model01, would it be possible to lift this part out, and only include the buttons when the currently selected keyboard is a Model01? Or better yet, only when the layout file exists?

File: src/renderer/screens/Editor/ImportExportDialog.js [hunk 2]
```diff
@@ -65,6 +85,10 @@ export const ImportExportDialog = withSnackbar(props => {
       });
   }
 
+  function loadDefault(layout) {
+    setData(JSON.stringify(layout));
+  }
```
Comment (algernon) [inline src/renderer/screens/Editor/ImportExportDialog.js]: To be future-proof, would it be possible to load from a file instead? Like, automatically construct the filename based on the active keyboard (a'la the default/experimental firmware filenames)? That way, we wouldn't need to import them directly, and would make it easier to add defaults for other keyboards than the Model01 later down the road.

Comment (TreTuna) [reply src/renderer/screens/Editor/ImportExportDialog.js]: Hmm... This does give me another idea for this that will integrate a solution for both of these comments. Will work on this and update PR when it's ready.

**Comment by TreTuna** [comment]
This will now look at the static directory for the vendor/product for Qwerty/Dvorak/Colemak JSON files. if they exist, it will show only links to load the default JSON data that it finds. (ie: if it only finds Dvorak, only a Dvorak link will appear)

I figured this was a start and if we wanted to add different default layouts we could either refactor it to load all from a specific _layouts_ directory for that board or just add them to the _defaultLayouts_ array in the _LoadDefaultKeymap_ component.

File: src/main/index.js [hunk 3]
```diff
@@ -29,6 +29,8 @@ let mainWindow;
 
 async function createMainWindow() {
   const window = new BrowserWindow({
+    width: 1200,
+    height: 900,
```
Comment (algernon) [inline src/main/index.js]: Could this be moved to a separate commit, or dropped from the PR? Window size seems irrelevant for clipboard stuff.

File: src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      fs.accessSync(path);
+      deviceLayouts.push({ name: layout, path });
+    } catch (err) {
+      console.log(`${vendor} ${layout} does not exist`);
+    }
+  });
+
+  return (
+    <div style={{ display: "flex" }}>
+      {deviceLayouts.length > 0 && (
```
Comment (algernon) [inline src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js]: Would it be possible to return an empty component if `defaultLayouts` is empty? Returning null before this main return seems cleaner than the condition here, and would result in an empty div less for boards that have no default layouts.

File: src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        <React.Fragment>
+          <h3>Load a default:</h3>
```
Comment (algernon) [inline src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js]: This would need to be localized. The Hungarian translation is: "Alapértelmezett kiosztás betöltése:" :)

File: src/main/index.js (hunk 3 above)
Comment (TreTuna) [reply src/main/index.js]: Ha! This actually wasn't even meant to go into this one, think it slipped in when I had a `git commit --amend --signoff` I'll pull it back. I was going to see about this though as I think the current default window size is pretty tiny, I always have to make it larger every time I open Chrysalis.

File: src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js (hunk 5 above)
Comment (TreTuna) [reply src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js]: Doh! Of course! Meant to do that.

File: src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js (hunk 4 above)
Comment (TreTuna) [reply src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js]: I believe the original reason I did this was that we may have to adjust the layout in the Dialog component if I returned null in case of empty. This is because we are currently using Flexbox `space-between` to evenly distribute this component apart from the clipboard links and put each to the left/right of this row. I could be completely wrong since I'm talking about CSS off the top of my head. 😂

File: src/main/index.js (hunk 3 above)
Comment (algernon) [reply src/main/index.js]: It might be worth saving the window size, position, etc and restore that on startup? But lets discuss that in a real ticket on its own :)

File: src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js (hunk 4 above)
Comment (algernon) [reply src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js]: Oh, right, fair. Lets stick to div + condition then, to err on the safe side.

File: src/main/index.js (hunk 3 above)
Comment (TreTuna) [reply src/main/index.js]: Will open this up now.

Comment (TreTuna) [reply src/main/index.js]: Reverted that change

File: src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js (hunk 4 above)
Comment (TreTuna) [reply src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js]: Just played with it, this would cause some additional complexity to make the layout work better as we'd have to check for the existence in the parent component and not in `LoadDefaultKeymap` and have a conditional that would modify the CSS there to take care of having a consistent layout.

File: src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js (hunk 5 above)
Comment (TreTuna) [reply src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js]: Got this, do we need to translate "Qwerty", "Dvorak", and "Colemak" or are we considering that universal?

Comment (algernon) [reply src/renderer/screens/Editor/ImportExportDialog/LoadDefaultKeymap.js]: I'd consider them universal.

**Comment by TreTuna** [comment]
Alright, I believe that's all resolved.

**Comment by algernon** [comment]
Sweet, will have a look after our nightly routine, in a few hours. Thanks!
